numpy/core/shape_base.py

```diff
@@ -423,32 +423,70 @@ def format_index(index):
         return parent_index, _nx.ndim(arrays)
 
 
-def _block(arrays, max_depth, result_ndim):
+def _assignment_recursion(result, arrays, list_ndim, start_indicies,
+                          depth=0):
+    if depth < list_ndim:
+        concat_depth = depth + result.ndim - list_ndim
+        for arr in arrays:
+            shape = _assignment_recursion(result, arr, list_ndim,
+                                          start_indicies, depth=depth+1)
+            start_indicies[concat_depth] += shape[concat_depth]
+        # lists are mutable, so we have to reset the start_indicies list
+        start_indicies[concat_depth:] = (0, ) * (result.ndim - concat_depth)
```

Comment:
A better way of handling this loop - initialize `start_indicies=()`, and do:
```
i = 0
for arr in arrays:
    shape = _assignment_recursion(result, arr, list_ndim,
                                          start_indicies + (i,), depth=depth+1)
    i += shape[concat_depth]
```

Reply:
Also, typo: `indices`, not `indicies`

Reply:
ok

Reply:
Thats not a typo, I just can't spell :+1:

Reply:
Fits into the broader category of failing to convert a sequence of phonemes in your head into a sequence of characters in memory, which I'm happy to write off as a typo :)